### DISCUSSION TITLE
Suggest a feature

### DISCUSSION THREAD

**Discussion by joebudi** [Budibase/budibase#1176] [Ideas]
Hey!

If Budibase does not support a feature you need, please detail it in the comments below. Please review comments before submitting to avoid duplication. To add weight to suggested features, you can upvote them with a 👍 .

Thank you!

**Comment by odysseyskate** [comment]
Provide a way to remove columns from a grid in design view (not database) e.g. auto ID

**Reply by joebudi** [reply]
This is coming in the next release!

**Reply by shishir2019** [reply]
Hi, has this been out now?

**Reply by LauridsJMikkelsen** [reply]
Also keen to know!

**Reply by v3ss0n** [reply]
Not avaliable yet , its needed especially when i just want to export a few column , not all 20 of them.

**Comment by odysseyskate** [comment]
Provide a way to increase the size of form text fields in height

**Comment by odysseyskate** [comment]
Provide a way to configure database options in design view e.g. rename an option, add or remove an option

**Reply by joebudi** [reply]
Great suggestion!

**Comment by odysseyskate** [comment]
Provide a way for cards to directly reference tables without needing a Repeater

**Reply by joebudi** [reply]
Sadly, this will not be possible. We are updating the repeater component to give it more power and functionality - like filtering, sorting, and more. It's an integral component, and hopefully, with its new functionality, you will love using it.

**Comment by odysseyskate** [comment]
Provide a way to make form fields transparent (no white background)

**Comment by odysseyskate** [comment]
Provide a way to count user-created database rows as an output for cards & other elements or allow elements to reference data views

**Comment by shaikhashraf709** [comment]
Queries on internal tables with "Where" clause I am badly in need of this feature. and i think many will agree

**Reply by shogunpurple** [reply]
Hi @shaikhashraf709 - this one is on the roadmap. You will be able to bind internal data sources to components the same way that you can bind external ones, allowing you to pass parameters for similar functionality of a where clause. You can track this issue here:

https://github.com/Budibase/budibase/issues/1110

In the interim - you can use a repeater to do this. You should be able to apply filters in the repeater, which should give you the functionality that you need.

![Screenshot 2021-03-03 at 10 22 36](https://user-images.githubusercontent.com/11256663/109791458-615e0180-7c0a-11eb-80e8-8d3a1fffc87b.png)

**Comment by thekingofspain** [comment]
- Create Table from Template (Custom Auto Columns)
- Table Types
-      RunTime Table (Existing functionality), 
-      Read Only Tables (Only editible in the IDE, Static/Enumeration based key/value pairs with optional user metadata columns))
-               Custom Template
-                       New Default Columns
-                                Name
-                                Short Name
-                                Active (display in the application dropdown for edit and new references)
-                                SortOrderValue  (default sort order of table values)
-                                Tooltip 
-                                Icon (Icon to reference)
-                                CssClasses
-                                ReferenceUrl

**Reply by joebudi** [reply]
Hey @thekingofspain (awesome name!). 

in regards to the read-only tables - is using a repeater and card a solution? 

I'm not sure what you mean by default template. Do you request a default template for default columns in the data section of Budibase?

**Comment by dantosXD** [comment]
I have a few Column requests:

1. A rating or scale column similar to this: https://support.airtable.com/hc/en-us/articles/360042311994-Rating-field#:~:text=The%20rating%20field%20allows%20you,from%20the%20default%20yellow%20star.
2. A slider column where you can choose the start and end values and choose either number or percent.
3. A formula column so I can do something like First Name Column + Last Name Column and have the row auto-populate with first+last

**Reply by joebudi** [reply]
Thank you @dantosXD . These are valid requests and I will note these requests to the team at our next roadmap meeting.

**Reply by gaa23** [reply]
I agree, really would like a slider with # or %

**Reply by AnnaJoker** [reply]
I am looking for a slider in budibase - could anyone update me here?

**Reply by melohagan** [reply]
You could try the *Super Slider* custom component: https://github.com/poirazis/bb-component-SuperSliderField

More on plugins here: https://docs.budibase.com/docs/custom-plugin

**Comment by dantosXD** [comment]
I would also love more charts but specifically a Radar or Spider Chart. 

Example: https://apexcharts.com/wp-content/uploads/2018/05/basic-radar-chart.png

**Reply by joebudi** [reply]
I assume you would need this feature, and the above "slider" data type request to be developed and released to solve your problem?

**Reply by gaa23** [reply]
It would for me. Slider with # or %

**Reply by fabriziosolaris** [reply]
More chart will be really a great enhancement

**Comment by dantosXD** [comment]
I would like the ability to edit the "Create New Row" popup so I can reorder the fields 

![image](https://user-images.githubusercontent.com/61808248/112143243-0175cd80-8bae-11eb-9a51-0851612f0d3a.png)

**Comment by EDMJohnny** [comment]
Custom calculations would be very helpful.   Currently you only allow Sum or Count.    To be able to do some custom math like multiply or divide two number columns - like quantity multiplied by price to get a total value

Looks like a very promising platform though.  Thanks

**Reply by ddune** [reply]
I can't build the "Estimator" app I want to without a "Calc" data type to perform basic math functions between two or more fields.

**Reply by joebudi** [reply]
Hey @EDMJohnny and @ddune. A calc column would be a great addition. In the meantime, Handlebars (accessible in the design section, within the binding interface) can help with some basic math:
https://www.youtube.com/watch?v=Ff_ceH31XAw

**Comment by llasaka** [comment]
Hello, I would like to show a message when a specific text field was found in a field of a table. I realised how to tell when it was not found, in "Empty text" property. But did not find how to trigger a "showmessage" action when a certain condition is true. Thank you

**Reply by joebudi** [reply]
Hey @llasaka . Do you want to display a list of information, that meets a certain requirement, from your tables?

Or do you want to display a component if X is true?

**Reply by llasaka** [reply]
The idea of the app is to check if a certain text is found in a list of items. It may trigger a message if there is a coincidence. You can check my progress in the live app https://app_bf7f96bb8a6b41fd9f742bd85d1b1230.app.budi.live/app_bf7f96bb8a6b41fd9f742bd85d1b1230#/

**Reply by llasaka** [reply]
as you can see, if you enter any random text in the upper field text, it tells there is no coincidence, I can do this with the "Empty text" field. If you enter 2876566 that is one of the items in the table, no message is shown, and I want to alert the user that there is a match

**Reply by llasaka** [reply]
also I would like to deploy this for a large group of job partners, so a universal user would be great to be created, with no need to enter a certain email address

**Reply by joebudi** [reply]
hey. I cannot check your app as it is private.

**Reply by llasaka** [reply]
Ok, but I think you can figure out what I am trying to do. It is so simple, but I don't know how to do it

**Comment by thabolowcode** [comment]
1. Have the ability to select which user to preview the app as... "Preview as Basic" or "Preview as Public" without having to log into the app to see how each user will experience the app.
2. Have the ability to choose who can see certain components on a specific page. For instance, ADMIN user can see the "Create New" button but a PUBLIC user can't.
3. On the budibase app, i can click on a row column (with a relationship) and it will take me to the DETAILS page of that column but i can't seem to do that on the deployed app. That would be awesome if i can do that on a deployed app.

**Reply by mjashanks** [reply]
Great suggestions, they all make alot of sense - thank you!

**Comment by clozach** [comment]
**Suggestion**
In the Builder, change "Zoom In" like so: `⌘+` → `⌘=`

**Rationale**
Unless one happens to use a number pad for this sort of thing, `⌘+` is actually just shorthand for `⇧⌘=`, so, operationally, the current Zoom shortcuts are actually `⇧⌘=` and `⌘-`.

`⌘+` and `⌘-`, offer the _illusion_ of symmetry at the _expense_ of operational symmetry.

**Reply by shogunpurple** [reply]
@clozach thank you for your suggestion. We will update the erroneous shortcuts to reflect this.

**Comment by ekansh005** [comment]
Hi,
I just signed up and was exploring the builder. There is no tooltip in most of the controls to suggest what it does. For example, I see a recycle kind of icon but I don't know what it does so a tooltip would be a good addition.
BR, Ekansh

**Reply by shogunpurple** [reply]
Hi @ekansh005 - thanks for your feedback!

This is in the pipeline for this month - we will be adding tooltips around the builder to make this easier for users.

https://github.com/Budibase/budibase/issues/3691

Thanks!

**Reply by AndyGUK** [reply]
Great suggestion @ekansh005. Being able to add tooltips within the designer would also be amazing. Not sure if there's a temporary work around to be able to do this at the moment?

**Comment by micebrain** [comment]
Allow option picker to have the options source to be linked directly to a data source without having to create a separate data provider component for the lookup table. As multiple option pickers will result in many nested data provider in the form design.

**Comment by micebrain** [comment]
Allow dynamic filter to filter on relationship column.

**Reply by shogunpurple** [reply]
Thanks @micebrain - you can track this one here. 

https://github.com/Budibase/budibase/issues/2316

**Comment by micebrain** [comment]
Allow "templating of component" to create reusable component that can be placed on multiple screens design. E.g. a container element that displays statistic that can appeared in more than one screen. So instead of copying and pasting the design from one screen to another, would be nice to be able to create the design and saved it in a "template library" and reuse it.

**Reply by shogunpurple** [reply]
This is a great suggestion, and something that we have talked about to help speed up workflows. I can't give you an exact timeline, but I would like to think that we will do more research and development around this in the coming months.

**Reply by micebrain** [reply]
That would be really nice! Am anticipating that :)

**Comment by micebrain** [comment]
Allow to customise display name of columns in table. It is currently showing the actual column name.

**Reply by kasgithub8** [reply]
Yes, we still need this

**Reply by mjashanks** [reply]
coming in the next release

**Reply by melohagan** [reply]
This one is done. 
**Configure columns** -> Add the columns you want -> set any custom labels you want.
![Screenshot 2022-09-02 at 10 06 39](https://user-images.githubusercontent.com/101575380/188105407-5b5e6c3b-f500-4f81-b3c6-7cff7f80b930.png)

**Comment by kovrom** [comment]
Timestamp in users table when they last logged in / last activity? This way it could be easier to remove dormant users, etc.

**Comment by AndyGUK** [comment]
Making the elements panel horizontally scrollable would be amazing. I seem to be forever running into this issue where my elements become hidden due to having multiple nested objects within my design.

![image](https://user-images.githubusercontent.com/73335367/155301962-ab19d9d7-fb5f-42fe-876a-dd2683f19cd1.png)

**Reply by joebudi** [reply]
This is now complete.

**Comment by bobhilbig** [comment]
Setting up queries on a dataase involves a lot of scrolling.
Could you create a little button to collapse the Tables list (and the other headings too) so i can get to my desired area quicker.
![image](https://user-images.githubusercontent.com/85771390/155403332-222d843d-8175-43cc-8687-41687a8c7cde.png)

**Comment by seaelem** [comment]
If you want a new column of a new table to be a relationship type, I find I first need to click away to another table before coming back to the new table and creating the relationship column. Else the table dropdown is blank.

<img width="721" alt="image" src="https://user-images.githubusercontent.com/26319178/160273903-6fcf03dc-1114-48ce-a92b-25b60ac085b2.png">

**Reply by melohagan** [reply]
This one was fixed! https://github.com/Budibase/budibase/issues/5444

**Comment by seaelem** [comment]
An undo button when editing screen layouts ;)

**Reply by shogunpurple** [reply]
Hi @seaelem - Undo/Redo is on the roadmap for May! We look forward to shipping it.

**Reply by seaelem** [reply]
heh heh I look forward to being an expert user of it :D

**Reply by NicolasFabre81** [reply]
What about this UNDO feature ? I just lost a a big work with a copy/paste mistake :-o

**Comment by chadgriff** [comment]
The new table component is great but stops short of providing conditional formatting.  The ability to define a background color or font color based on the value of a cell within a table would be an awesome enhancement.

**Comment by dannyshin76** [comment]
It would be great to be able to specify date format setting (such as "MM-DD-YYYY") to:
1. Date Picker component in forms
2. Columns with date/time type in tables
3. Other components or elements where a date would be visible to users

**Reply by tvogt** [reply]
If you mean a global setting, I'm for it.

**Comment by dannyshin76** [comment]
My use case requires an UI to select a month, not a date. I would be great if Budibase provides a Month Picker component. Another approach would be to have a "Month Only" setting to the Date Picker.

**Reply by RSPont** [reply]
Make an enum (or options in budibase db)

**Comment by oitkenmen** [comment]


**Reply by RSPont** [reply]
you can always export the json and import it, but I agree, It's not really a great solution, while doable

**Comment by sid6p8** [comment]
"Singleton Provider"

It would be nice if you could create a combined data provider and repeater for queries that return a single item.

**Reply by shogunpurple** [reply]
This exists! It’s called the “Repeater Block” under the blocks section.

**Comment by melohagan** [comment]
At the moment, navigation doesn't work after a 'Close Screen Modal' action. Could be handy to add an optional redirect field to this action to allow a redirect on close.  

![Screenshot 2022-04-22 at 11 19 16](https://user-images.githubusercontent.com/101575380/164689276-16afc4e5-9b3e-4301-8e2f-84fe0af47cdb.png)

**Reply by ikaheraJQ** [reply]
Sorry @melohagan, when will this feature be available or released?

**Reply by melohagan** [reply]
Replied here by mistake: https://github.com/Budibase/budibase/discussions/5626#discussioncomment-2653269

**Comment by andyburgessmd** [comment]
It would be useful if a start and end date could be set on the date picker so that you can only select a date from within that range.

I'd use this to not allow dates in the past to be selected as an example.

**Reply by melohagan** [reply]
There is a date range component if that helps!

**Reply by kasgithub8** [reply]
> It would be useful if a start and end date could be set on the date picker so that you can only select a date from within that range.

It's literally right there?
![image](https://user-images.githubusercontent.com/95284278/186641530-846a3de2-0152-46c3-bcb7-a9ac2e1ceae8.png)

**Reply by andyburgessmd** [reply]
Sorry, maybe I should have said that can be applied to the data picker form component in the application design/front end and can be set by a binding. 
So those two earliest and latest values but where I could apply a binding like earliest = {{ date now }} and latest = {{ date "1 week from now" }} (though I'd do this in JS using moment)

**Reply by kasgithub8** [reply]
@andyburgessmd 
Ah, yes.
That's not a thing atm. but it's a good request that I can get behind.
JS is the way rn.

**Reply by andyburgessmd** [reply]
Whilst I am on the date picker - it would be great if you could set the week start day. At the moment its the US format where week starts on Sunday whereas the rest of the world and ISO use Monday.

**Comment by tvogt** [comment]
When using an external database that has foreign keys set up (like Postgres), auto-populate the relations when fetching the tables. Why do I need to set the exact same relations up again manually in Budibase? It's not just extra effort, there's also a risk of making mistakes.

**Comment by tvogt** [comment]
Please allow charts to configure min and max values for the axis. For example, I have a data set of percentage values, which can never go above 100 and yet Budibase automatically sets the max of the chart to 120, probably to have some extra space.

Also, allow a prefix and suffix to the values, % in my case, maybe € or $ for currencies, etc. - I know some convoluted CSS can probably accomplish that, but it should be available to non-techies.

**Comment by giacomouberti** [comment]
Add the option to select a destination for URLs in Navigation and Actions links at the moment it's possible only for links

**Comment by NicolasFabre81** [comment]
Add a good filter block that can be added to a table to get dropdown filters upon each column.

Each time I develop a list,  I have to re-make this kind of filters. 
The values in each dropdown list should be the distinct values of the field in the parent data provider.

<img width="599" alt="image" src="https://user-images.githubusercontent.com/28592069/191518364-10f32459-77f7-44cd-a1d2-60d9d1ad1045.png">

**Reply by joebudi** [reply]
Hey. I could be very wrong, but is this not possible with the table block?

**Reply by NicolasFabre81** [reply]
Somehow it is, but by default it gives you text fields, not dropdown fields. To get a dropdown, you have to use a BB intern table, and define your field as an option list. In my case I use a postgreSQL database with varchar fields so I don't get the dropdown filters.

But you're right, the solution is to improve the table block.

<img width="753" alt="image" src="https://user-images.githubusercontent.com/28592069/191525019-8e06da50-3932-4ef3-8836-584048c9417f.png">

**Reply by oitkenmen** [reply]
I guess the solution is to wrap the Table into the Form component. Then add Fields component above the Table. Inside the Field component, you can make dropdowns corresponding to the columns. Set the columns' width and the dropdowns' width accordingly. Connect the filters inside the Table to the dropdowns.

**Comment by oitkenmen** [comment]
![image](https://user-images.githubusercontent.com/3124697/204723530-65745551-a40e-489a-8af4-764fa4e8b7c6.png)

Let's make the Handlebars field monospaced, similar to JavaScript field.

**Comment by gmentzik** [comment]
Import CSV file to JSON Array of Objects. Allow this to or even better any array of json objects with same schema to be used as provider input. This will allow import or csvs to a DB table , at least row by row, as this is supported right now.

**Comment by gmentzik** [comment]
Store/Update multiple rows from forms in repeater with handlling errors options : transactional (reject all rows insert/updte in case of error), stop on error and inform, ignore error but inform user. None of these is provided by automations loop right now.

**Comment by gmentzik** [comment]


**Reply by giacomouberti** [reply]
If I understand correctly the request you can already selecting the action "External Source" and select one of the query mapped

**Reply by gmentzik** [reply]
> If I understand correctly the request you can already selecting the action "External Source" and select one of the query mapped

You are right. I removed this.

**Comment by GregFdz** [comment]
- Using automation as data provider. 
- More advanced conditionnals in automation. Example : add a "if" field to each step block.
- Letting users get the feedback from an automation.
- Hashing functions.
- Encrypting.
- Min length for columns and fields.
- Reordering columns (so the last column you create can be moved to the left).
- Being able to set a regex form validation without making it "required" (doable with a trick).
- Allowing more complex JS in automations.
- Editing global css style.
- Ability to add custom header / footer content.
- Stopping automation to B step if A failed when B is a backend log.
- Naming automation steps and using names instead of step number.
- Or at least displaying the steps' number. This one can be done with just a few lines of CSS.

**Reply by melohagan** [reply]
Thanks for the ideas! 
The first one probably wont ever happen because the automations run asynchronously in the background, i.e. fire and forget. 
Perhaps in future automation events could be added, e.g. on complete push a notification to a user.

**Reply by GregFdz** [reply]
What if every user had a session and automation feedbacks were temporarily stored in a log table with a hash of the session ? and maybe even destroyed as soon as the user reads them for the first time. I'm actually trying to achieve it from my side but that's way more dirtier than if you guys (probably much more skilled than me) were to build this at a core level. I'd love to help open source projects like yours but as a self taugh programmer I always feel like I lack so much knowledge and skills... I can only understand things and make suggestions 😅

**Comment by gmentzik** [comment]
Add property to form fields to be able to check if value changed compared to initial/default value. This will allow to enable buttons or add css to fields value is modified. Feature could also be added to Form to check if any of fields value is altered compared to default or initial that could be used to enable save button or used at any other condition in page.

**Comment by GregFdz** [comment]
I forgot that one : 
- Add "goto" in automations.

I think that the ideas I have for automations ( "if" and "name" fields for each step block an goto) are much less efforts than actually building a complex conditionnal GUI (and actually building the logic of it). I'd say that, even without having dived in budibase's source code yet, I think it could probably easily be added to the existent code.

Nb : the "goto" would be a step block of itself with a "where" an if and a (required ?) "limit" field. The "limit" would allow to count the number of iterations if the goto sends the cursor before his own placement in the logic line, thus preventing infinite loops. Last idea about this : maybe one more field to return a value if the limit has been reached.

**Reply by IchthysMaranatha** [reply]
Here https://github.com/Budibase/budibase/discussions/10123
In actions and automations

**Comment by AlejandroFNadal** [comment]
Hello everyone!

- Trigger notifications from javascript code, or otherwise, add a condition option in the notification block.
- Allow to trigger automations when a new user is created. Right now I have an internal coworkers table and the budibase table of users and I want to add a key from the budibase table to the coworkers table. 
- For form validation, add an option that just runs a JS function and if the function returns false, declare the value to be invalid. Specially for number fields, as with text fields we already have regex matching. (what I am doing now is using text fields everywhere and converting the values to integers)

**Comment by muksidin** [comment]
Provide a way to create report as pdf.

**Reply by melohagan** [reply]
A very popular request! https://github.com/Budibase/budibase/issues/1372

**Comment by oitkenmen** [comment]
Hire a Budibase Developer. I think it's time to create this kind of a page on the Budibase website. Similar to [Webflow](https://experts.webflow.com).

**Reply by oitkenmen** [reply]
A showcase page. Built on Budibase. Made in Budibase. Again, similar to [Webflow](https://webflow.com/made-in-webflow).

**Reply by joebudi** [reply]
This is a great idea and something we want to do!

**Comment by benaunt** [comment]
Ability to generate one or more email address for each app that can receiving incoming emails then grab any attachment, subject, body etc and store in a database.

It would be great if it allowed you to point what database/table each incoming email points to, even tables outside the base budibase database.

**Comment by dannyshin76** [comment]
1. Add "Is truthy" and "Is falsey" to basic condition evaluators. Obviously they won't need subsequent clauses.
![image](https://user-images.githubusercontent.com/93232582/207511306-cb48053c-c1ba-4e3b-8854-f5eb094f399a.png)

2. Add an "empty" data provider without actual data source, which will act like a "for loop" by using the Row Index. It would be best if the "limit" can also be conditionally specified using handlebar or JS.

**Reply by andz-bb** [reply]
@dannyshin76 for your second idea, a Budibase community member has recently built a custom component plugin that acts as an unbound repeater. It's worth checking it out to see if it fits your use case: https://github.com/poirazis/bb-component-SuperRepeater

**Comment by benaunt** [comment]
Another useful addition would be user interaction actions to the workflow system along with back end processes for managing them.

Key actions could include:

- Get data from user - allows a custom budibase form to be linked to the action that maps to a database table, the form should be customisable using the normal budibase features and be able to display data based on the content the workflow is running against. Eg passing in parent record data etc.
- User approval action - a simplified version of the get data from user allowing for a drop down with the default approve/rejected values and a comment box. The drop down values should be able to be changed or additional options added.

Both actions should allow for:

- Reminders to be specified
- Auto escalation pathways or exit points to be specified if the assigned user(s) do not action within a set timeframe
- User specified delegation if the assigned user wants to pass the action on
- Security so only the assigned user(s) or admins can perform the action
- The user forms for each action should run in the UI of the related budibase app
- Custom email building with dynamic properties of the URL to the action/form and properties to the data gathered in the form
- Upon completion it should also return whom completed the action, the date of completion
- These properties should be available for subsequent workflow actions to use for processing or saving to a database etc
- The ability to specify where a user will be directed to after completing an action eg the apps home screen, a completion screen or something else
- When setting who an action is assigned to, it should allow for one or more assignees. When its multiple assignees have settings to specify if all people need to respond, completes on the first response. The options to set the delivery method so that it goes out to everyone all at once or loops through each assignee individually.

The budibase system should also have a universal feature to manage these types of actions

- A centralised screen where an individual logged in user can view all pending actions assigned to them across all budibase apps on the instance
- The ability for a user to specify an away status with a date range and to set a delegate, if anything is assigned to them in this period it auto delegates
- API's to this data allowing it to be pulled to other systems, or even into budibase apps



Thanks :)

**Comment by benaunt** [comment]
Have another one that is more useful for large organisations.

Once budibase has been configured for SSO, the ability to utilise your SSO providers user groups to grant access to budibase apps instead of manually adding users would be hugely beneficial.

This will allow improved security as managing access to multiple IT systems outside of budibase can be maintained using those existing groups. This will reduce issues where staff have left or moved teams and the wider organisational groups have been updated but not access within budibase. It also allows more automation of access for organisations that have that benefit. 

Possibly this could work by:

- Within the existing budibase groups feature, allow the registration of one or more SSO groups
- Once a group is registered, budibase automatically pulls in all users of that group to the budibase group
- Once a day budibase automatically resyncs the SSO provider group with its own group and removes/add users as required
- Maybe the ability to manually force a resync as needed

Thanks

**Reply by dewdeded** [reply]
I think they work on that.

**Comment by BaNkIck** [comment]
Provide a way to dynamically change the sorting of a repeater by the user.

Currently, it's only possible to sort data by using a table component. The repeater is much more powerful as it allows the creation of custom and more complex designs to display each element of the database. However, there's no way to add a dropdown or other similar control to allow the users to choose how they want to order the elements.

**Reply by BaNkIck** [reply]
I'd like to clarify that I meant the Data Provider of a repeater. I realize the repeater is not the component doing the sorting.

Additionally, it'd be great (and perhaps it's easier and can be done as a "first step") to completely disable the sorting of the data provider. After testing, it seems that it's always sorting the results, even when nothing is selected on the "Sort Column" option. In this case, I think, it sort by the first field of the results.

I managed to add the sorting to the query via a transformer (using MongoDB as the datasource) but now the Data provider is re-sorting the results anyway.

**Comment by BaNkIck** [comment]
Improve pagination of data views by adding the total count of rows on each table, faster pagination browsing (directly going to a specific page, or the last one), and the ability to choose how many records per page to show.

**Reply by L42xyz** [reply]
Hi, 
When biding to a data provider, it would be nice to have the total rows length and not only the paginated rows length 

<img width="405" alt="image" src="https://github.com/user-attachments/assets/171f5ee2-3971-44f5-bad8-e3fdb26a327f">

Currently, I have to create 2 queries (one with pagination and the other one without).

Many thanks!

**Comment by BaNkIck** [comment]
Provide a way to "truncate" (delete all records) a table.

**Comment by BaNkIck** [comment]
Provide a way to update records if they already exist while importing from a CSV file instead of adding everything as new rows.

The match could be done by selecting a field to use as the key.

**Reply by melohagan** [reply]
This would be a nice one! https://github.com/Budibase/budibase/issues/7400

**Comment by BaNkIck** [comment]
Add bulk operations of the database to the automation actions.

Currently, there are only actions to update or delete a single record which can, sometimes, be very slow. Having actions to massively update or delete all the records that match certain conditions would greatly improve performance.

**Reply by giacomouberti** [reply]
I'm currently achieving this using a custom query with some bindings in order to select which record to update or delete and to call it as an external source from the automation.

**Reply by BaNkIck** [reply]
This is a great idea! I'll use it, thanks!

However, this would not work when using the internal tables as they don't support custom queries. Also, I think the idea of Budibase is to be low-code and, even that this is a possibility and it's nice to have for more advanced users, I'm sure it fits under the low-code idea.

**Comment by BaNkIck** [comment]
Allow placing child components of tables at any column.

Unless I'm missing something, child components added to a table can only be placed at the beginning of it. It would be great to be able to place them anywhere. Perhaps, the components could be displayed in the columns manager and use this to order them between any other columns.

**Comment by jdids247** [comment]
Allowing click and drag resizing of column widths similar to Excel. After an update, all of our fields are unable to size correctly using "Auto". Changing the width creates vertical view as discussed in this issue: https://github.com/Budibase/budibase/issues/5348

We're using MSSQL in this example.

![image](https://user-images.githubusercontent.com/110419431/228887337-82a2babf-bea2-42d6-8408-110b94ff96e2.png)

**Reply by melohagan** [reply]
Hey @jdids247 

You might be interested in trying out the Super Table! It's a community [custom component](https://docs.budibase.com/docs/custom-component) plugin.


https://user-images.githubusercontent.com/101575380/228893691-db9c0b03-2ddf-482a-bd7a-4ed181bbde81.mov

**Comment by Albatrossoph** [comment]
I'd be interested in having the option of hiding a button (or anything with conditional UI) after or before a **configured** date. It is already possible to select a date 'by hand'. But I want to have the option via a binding. I am using a configuration table where I store start and end date for a registration process. But as soon as i want to link this date via binding to my conditional UI, I loose the specific date options 'More than' or 'less than' and thus cannot use it in the way I want to...

**Reply by kasgithub8** [reply]
Is this not possible via JavaScript?

**Comment by gmentzik** [comment]
There is a new feature at Forms called Read only.
It would be nice if there was a Button action to set Form and Form Fields  Read Only and Disabled state instead in additional to already existing Component Conditional functionality.

**Reply by melohagan** [reply]
It would certainly make things easier, but you can technically do with the **Update State** action, and then have a condition on that app state.

There's a feature request here: https://github.com/Budibase/budibase/issues/7979

**Comment by gmentzik** [comment]
It would be useful if we could export a screen to reuse it by importing it on another application. Yes it would break the relationships with DB but we would have the structure and design ready, so the only step would be to go through components and fix relations. Now we have to create a page from scratch.

**Reply by sdsys-ch** [reply]
I agree and would add the ability to export and import queries by data source or data sources with their queries as defined in another app. Same goes for actions.

I often find myself prototyping in a test application only to have to rebuild everything by hand on the production system.

The alternative might be to build / update in a test environment including all "snippets" (queries/ screens/ ...) and duplicate that once a new version is done, delete all unwanted "snippets" from the duplicate, then export that and import on production system to replace the old one, I guess (highly error prone). That doesn't work well for disparate apps though, and the idea might be to have a "master app" including all screens/ data sources / actions/ ... that is duplicated for every new app and pruned for necessary functionality. Not ideal either, I believe.

While probably not easy to implement, it should be feasible, no? For queries this could be "Import Queries->Attach to datasource?" etc. and similar for screens that contain data sources that aren't referenced (yet) in the current app?

**Reply by joebudi** [reply]
We plan to release Workspaces next week. With Workspaces, you can share data sources and components among apps.

**Comment by andyburgessmd** [comment]
Shared/global tables would be really useful. An example is an assets table that could be used across applications.

**Comment by gmentzik** [comment]
We need a way to load a csv file in frontend , manipulate the data and then store to a data (eg. mysql) table. 
The current custom csv_import plugin does not fit the purpose as it directly writes to table and we cannot process the internal json array it creates (data variable). The most convenient would be a long text field with a button to load csv file and present the json array result. Then we could take the result and use it as input to provider/automations/queries etc...
Another good idea would be to allow a way have an option to feed a table block with a csv and present data for the Frontend.
What do you think?

**Comment by fabriziosolaris** [comment]
In case of single select Columns, or in multiselect, it should be nice to specify custom colors

**Reply by joebudi** [reply]
This has been requested a few times. We will expand the color set in a future release.

**Comment by fabriziosolaris** [comment]
As in previous table, permit to add icons and Images, links an so on, in columns

**Comment by IchthysMaranatha** [comment]
Forms generated from schema received through api or using custom json
eg https://www.airwallex.com/docs/api#/Payouts/Beneficiaries/_api_v1_beneficiary_api_schemas_generate/post
https://www.airwallex.com/docs/api#/Payouts/Beneficiaries/_api_v1_beneficiary_form_schemas_generate/post
There seems to have been quite a few requests earlier also like
https://github.com/Budibase/budibase/issues/10860

**Reply by joebudi** [reply]
Thanks for the request. This is great because we will improve form building and automations over the next few cycles.

If you can, would you be happy to share more details re your use cases/requirements via a quick video chat? You can book time in my Cal using the link below:
https://cal.com/joe-johnston/15min

Or send me an email: 
joe at budibase dot com

**Comment by zan-xhipe** [comment]
Add the ability to validate a field based on a REST API call.

**Reply by melohagan** [reply]
You can do this sort of thing: https://docs.budibase.com/docs/lookup-record

Make a query on button click, or field change.

**Comment by fabriziosolaris** [comment]
We use budibase from a long time.
In previous version of table, there are a lot of features that has been lost in actual revision.
Actual table is worst and we i start a new project i clone a old project with good table, delete all and start the new app.
We are evaluating to swap bb for the limits of this roadmap.

**Reply by Stefano-Paul-Pascu** [reply]
I completely agree; the old table was much more functional and complete, even aesthetically. It should also be possible to choose whether to make the first column static or not. Another feature that should be included, in my opinion, is the ability to filter directly from the table using an options picker when the column accepts enum data (similar to the super table). Finally, the button management should be more similar to the old table, placed unobtrusively on the left rather than on the right as is currently the case. In this regard, it would also be ideal to be able to insert icons (in an optimal manner) as buttons, as in the old table.
